src/RequestHandler/outofprocess/serverprocess.cxx

```diff
@@ -1159,6 +1170,9 @@ SERVER_PROCESS::CheckIfServerIsUp(
             goto Finished;
         }
 
+        // More entries may be created between the two GetExtendedTcpTable calls
+        // Give a bigger buffer, 200 byte is big enough to hold additional entries
+        dwSize += 200;
```

Comment:
I don't like the approach here. Why can't we just call GetExtendedTcpTable with an initial size for the pTCPInfo, and keep expanding it until the call succeeds?

Reply:
we don't want to be in the situation of infinite loop (multiple tries) if processes continue open ports.

Reply:
1. We can limit the number of retries we do to a fixed number.
2. Can processes opening ports out grow like a 2x buffer size increase? I doubt it.

Reply:
Yeah, let's avoid the magic const and do the loop

Reply:
You don't want waste CPU cycles on those retries. Why not give a bigger enough buffer ？

Reply:
You don't want waste CPU cycles on those retries. Why not give a bigger enough buffer ？

Reply:
Make a retry count 2, just write it as a loop and double the buffer size. I would be much cleaner.

Reply:
Because this can still fail.  I'd rather guarantee reliability here rather than worry about why a process didn't start randomly.

Reply:
200 would only fit 10 entries. `MIB_TCPROW` is 20 bytes.